### DISCUSSION TITLE
Extending intializationOptions with which eclipse.jdt.ls starts

### DISCUSSION THREAD

**Discussion by zytact** [mfussenegger/nvim-jdtls#139] [Q&A] [answered]
I am very new Lua. I hope this isn't too much of a bother. I couldn't figure out where to place the code for [java-debug](https://github.com/mfussenegger/nvim-jdtls#java-debug-installation) and for [nvim-dap-setup](https://github.com/mfussenegger/nvim-jdtls#nvim-dap-setup). I'm pretty sure I'm missing something. Thank you.

**Comment by EpsilonKu** [comment, marked as answer]
U need write it before `require('jdtls').start_or_attach(config)`. Set config like array or object ([example](https://github.com/mfussenegger/dotfiles/blob/89a0acc43ac1d8c2ee475a00b8a448a23b8c1c26/vim/.config/nvim/lua/lsp-config.lua#L126-L196)). 
```
local config = {}
local workspace_folder = home .. "/.workspace/" .. vim.fn.fnamemodify(root_dir, ":p:h:t")
config.init_options = {
  bundles = {
    vim.fn.glob("path/to/java-debug/com.microsoft.java.debug.plugin/target/com.microsoft.java.debug.plugin-*.jar")
  }
}
config.cmd = {"jdtls.sh", workspace}
config.root_dir = require("jdtls.setup").find_root({"gradle.build", "pom.xml"})

require ('jdtls').start_or_attach (config)
```
Hope u understand.

**Reply by zytact** [reply]
Got it! Thanks a lot!
